featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,261 @@
+
+    Args:
+        es (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `es`.
+        is_time_series (bool): Whether or not time-series analysis will be performed. If set to `True`, `Lag` will always be recommended,
+        as well as all Rolling primitives if numeric columns are present.
+        excluded_primitives (List[str]): List of transform primitives to exclude from recommendations.
+
+    Note:
+        This function currently only works for single table and will only recommend transform primitives.
+    """
+    recommended_primitives = set()
+
+    if not is_time_series:
+        excluded_primitives += TIME_SERIES_PRIMITIVES
+
+    all_trans_primitives = get_transform_primitives()
+    selected_trans_primitives = [
+        p for name, p in all_trans_primitives.items() if name not in excluded_primitives
+    ]
+
+    valid_primitive_names = [
+        prim.name
+        for prim in get_valid_primitives(
+            es,
+            target_dataframe_name,
+            1,
+            selected_trans_primitives,
+        )[1]
+    ]
+
+    recommended_primitives.update(
+        _recommend_non_numeric_primitives(
+            es,
+            target_dataframe_name,
+            valid_primitive_names,
+        ),
+    )
+    skew_numeric_primitives = set(["square_root", "natural_logarithm"])
+    valid_skew_primtives = skew_numeric_primitives.intersection(valid_primitive_names)
+
+    time_series_primitives = set(TIME_SERIES_PRIMITIVES)
+    valid_time_series_primitives = time_series_primitives.intersection(
+        valid_primitive_names,
+    )
+
+    numeric_tags_only_df = es[target_dataframe_name].ww.select("numeric")
+
+    if valid_skew_primtives:
+        recommended_primitives.update(
+            _recommend_skew_numeric_primitives(
+                numeric_tags_only_df,
+                valid_skew_primtives,
+            ),
+        )
+
+    if valid_time_series_primitives:
+        recommended_primitives.update(valid_time_series_primitives)
+    return list(recommended_primitives)
+
+
+def _recommend_non_numeric_primitives(
+    es: EntitySet,
+    target_dataframe_name: str,
+    valid_primitives: List[str],
+) -> set:
+    """Get a set of recommended non-numeric primitives given an entity set.
+
+    Description:
+        Given a single table entity set with a `target_dataframe_name` and an applicable list of `valid_primitives`,
+        get a set of primitives which produce non-unique features.
+
+    Args:
+        es (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `es`.
+        valid_primitives (List[str]): List of primitives to calculate and check output features.
+    """
+
+    recommended_non_numeric_primitives = set()
+    # Only want to run feature generation on non numeric primitives
+    es_without_numerics = EntitySet("ES without numerics")
```

Comment:
Instead of making a second entityset, we could use the `ignore_columns` parameter in `dfs` to filter out all numeric columns for that `dfs` run

There's an issue with how the second ES is created that we can avoid by not making a second one

Reply:
ahh nice yeah that would work better